detection_rules/version_lock.py

```diff
@@ -4,19 +4,80 @@
 # 2.0.
 """Helper utilities to manage the version lock."""
 from copy import deepcopy
-from typing import List, Optional
+from dataclasses import dataclass
+from pathlib import Path
+from typing import ClassVar, Dict, List, Optional, Union
 
 import click
 
+from .mixins import LockDataclassMixin, MarshmallowDataclassMixin
 from .rule_loader import RuleCollection
+from .schemas import definitions
 from .semver import Version
-from .utils import dict_hash, load_etc_dump, save_etc_dump, cached
+from .utils import cached, get_etc_path
 
 ETC_VERSION_LOCK_FILE = "version.lock.json"
+ETC_VERSION_LOCK_PATH = Path(get_etc_path()) / ETC_VERSION_LOCK_FILE
 ETC_DEPRECATED_RULES_FILE = "deprecated_rules.json"
+ETC_DEPRECATED_RULES_PATH = Path(get_etc_path()) / ETC_DEPRECATED_RULES_FILE
 MIN_LOCK_VERSION_DEFAULT = Version("7.13.0")
 
 
+@dataclass(frozen=True)
+class VersionLockFileEntry(MarshmallowDataclassMixin):
+    """Schema for a rule entry in the version lock."""
+    rule_name: definitions.RuleName
+    sha256: definitions.Sha256
+    type: definitions.RuleType
+    version: definitions.PositiveInteger
+    min_stack_version: Optional[definitions.SemVer]
+
+    # TODO: need to exclude nested 'previous'
```

Comment:
You planning to do this in this PR?

Reply:
good question ...

Reply:
I went ahead and created an issue so that we can close this: https://github.com/elastic/detection-rules/issues/1930 When you get a chance can you update the description in case it's missing anything?

Reply:
thanks, will address in separate PR